### PR TITLE
Add Search.repository endpoint

### PR DESCRIPTION
**Pull request by timo-schmid**
Adds a new endpoint to search for repositories.

Some test code I've used to test:
```
package io.chrisdavenport.github

import cats.Show
import cats.effect.{ExitCode, IO, IOApp}
import cats.implicits._
import io.chrisdavenport.github.data.Repositories.Repo
import io.chrisdavenport.github.data.SearchResult
import io.chrisdavenport.github.data.SearchResult.Order.{Ascending, Descending}
import io.chrisdavenport.github.data.SearchResult.Sort.{Stars, Updated}
import io.chrisdavenport.github.data.SearchResult.{Order, Sort}
import org.http4s.client.Client
import org.http4s.client.blaze.BlazeClientBuilder

object Search extends IOApp {

  val auth: Option[Auth] = Some(OAuth("[insert-api-key]"))

  override def run(args: List[String]): IO[ExitCode] =
    BlazeClientBuilder[IO](scala.concurrent.ExecutionContext.global)
      .resource
      .use(testRepositorySearch) *> ExitCode.Success.pure[IO]

  private def testRepositorySearch(client: Client[IO]): IO[Unit] =
    List[(String, Option[Sort], Option[Order])](
      ("svalidate user:timo-schmid", None, None),
      ("user:timo-schmid", Some(Stars), Some(Descending)),
      ("user:timo-schmid", Some(Updated), Some(Ascending))
    ).traverse_(repositorySearch(client))

  private def repositorySearch(client: Client[IO])(params: (String, Option[Sort], Option[Order])): IO[Unit] =
    endpoints.Search
      .repository[IO](params._1, params._2, params._3, auth)
      .run(client)
      .flatMap { result =>
        IO(println(result.show))
      }

  private implicit val showRepo: Show[Repo] =
    repo => s"- ${repo.updatedAt.get} ${repo.stargazersCount}★ ${repo.owner.login}/${repo.name}"

  implicit def showSearchResult[A](implicit showA: Show[A]): Show[SearchResult[A]] =
    searchResult =>
      (List(
        "Search Results:",
        s"- total_count: ${searchResult.totalCount}",
        s"- incomplete_results: ${searchResult.incompleteResults}",
      ) ++ searchResult.items.map(_.show)).mkString("\n")

}
```

### CODE REVIEW

File: core/src/main/scala/io/chrisdavenport/github/endpoints/Search.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  def repository[F[_]: Sync](
+    q: String,
+    sort: Option[Sort],
+    order: Option[Order],
+    auth: Option[Auth]
+  ): Kleisli[F, Client[F], SearchResult[Repo]] =
+    RequestConstructor.runRequestWithNoBody[F, SearchResult[Repo]](
+      auth,
+      Method.GET,
+      (uri"/search" / "repositories")
```
Comment (ChristopherDavenport) [inline core/src/main/scala/io/chrisdavenport/github/endpoints/Search.scala]: Didn't you modify us to use relative?

```suggestion
      (uri"search" / "repositories")
```

Comment (timo-schmid) [reply core/src/main/scala/io/chrisdavenport/github/endpoints/Search.scala]: Good thing you're reviewing :sweat_smile:

File: core/src/main/scala/io/chrisdavenport/github/endpoints/Search.scala [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import org.http4s.client.Client
+import io.chrisdavenport.github.Auth
+import io.chrisdavenport.github.data.SearchResult
+import io.chrisdavenport.github.data.SearchResult.{Order, Sort}
+import io.chrisdavenport.github.internals.GithubMedia._
+import io.chrisdavenport.github.internals.RequestConstructor
+
+object Search {
+
+  def repository[F[_]: Sync](
```
Comment (ChristopherDavenport) [inline core/src/main/scala/io/chrisdavenport/github/endpoints/Search.scala]: Can you link to the appropriate endpoint documentation for easy debugging in the future?

**Comment by ChristopherDavenport** [comment]
I would love a unit test on the endpoint generating the request you expect, but understand and don't always live up to that myself.

**Comment by timo-schmid** [comment]
Makes sense, I'll add one!

File: core/src/main/scala/io/chrisdavenport/github/endpoints/Search.scala (hunk 2 above)
Comment (timo-schmid) [reply core/src/main/scala/io/chrisdavenport/github/endpoints/Search.scala]: Yep, will do!

File: core/src/main/scala/io/chrisdavenport/github/internals/RequestConstructor.scala [hunk 3]
```diff
@@ -89,7 +89,7 @@ object RequestConstructor {
       .void
       .stream
 
-  private val GITHUB_URI: Uri = uri"https://api.github.com"
+  private val GITHUB_URI: Uri = uri"https://api.github.com/"
```
Comment (timo-schmid) [inline core/src/main/scala/io/chrisdavenport/github/internals/RequestConstructor.scala]: Turns out this didn't work with relative urls anymore (probably due to the missing slash).
Adding the slash at the end fixes it.

File: core/src/test/scala/io/chrisdavenport/github/endpoints/SearchSpec.scala [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+import org.http4s.implicits._
+import org.http4s.client._
+import org.http4s.circe._
+import org.http4s.dsl.io._
+
+class SearchSpec extends Specification with CatsEffect {
+
+  "Search" should {
+
+    "return a valid search result" in {
```
[the remark this answers is not in the document]
Comment (ChristopherDavenport) [reply core/src/test/scala/io/chrisdavenport/github/endpoints/SearchSpec.scala]: Something is better to nothing.

File: core/src/main/scala/io/chrisdavenport/github/internals/RequestConstructor.scala (hunk 3 above)
Comment (ChristopherDavenport) [reply core/src/main/scala/io/chrisdavenport/github/internals/RequestConstructor.scala]: kk. Expect that will then be required as well for enterprise versions of that uri. Things to note when we get a bug report.

Comment (timo-schmid) [reply core/src/main/scala/io/chrisdavenport/github/internals/RequestConstructor.scala]: Yes, in fact I remember I had to add the `/` at the end of the uri in the enterprise version to make it work.

**Pull request merged**